### DISCUSSION TITLE
Error: 'PacketCapture' does not contain a definition for 'Packet'

### DISCUSSION THREAD

**Discussion by lev99999** [chmorgan/sharppcap#436] [General]
Hello,

I use .net core 7 and I downloaded sharpPcap library to vscode using the command:
dotnet add package SharpPcap --version 6.2.5

The library was successfully installed, but when I try to use this command for example:
using var device = new CaptureFileReaderDevice("filename.pcap");
device.Open();
device.OnPacketArrival += Device_OnPacketArrival;
device.Capture();
(of course I added this: using SharpPcap).

I got this error:
'PacketCapture' does not contain a definition for 'Packet' and no accessible extension method 'Packet' accepting a first argument of type 'PacketCapture' could be found

Thank You

**Comment by kayoub5** [comment]
@lev99999 full code snippet please and full stack trace/line numbers

**Comment by lev99999** [comment]
**This is the code:**

```
using Microsoft.AspNetCore.Mvc;

using SharpPcap;



namespace MyProjectApi.Controllers;



[ApiController]

[Route("[controller]")]

public class PcapsController : ControllerBase

{

    public void Device_OnPacketArrival(object s, PacketCapture e)

    {

        Console.WriteLine(e.Packet);

    }



    [HttpGet(Name = "readPcaps")]

    public void readPcaps()

    {

        using var device = new CaptureFileReaderDevice("log111622.pcap");

        device.Open();

        device.OnPacketArrival += Device_OnPacketArrival;

        device.Capture();

    }

}


```

**And this is the results of building:**

SBuild version 17.4.0+18d5aef85 for .NET
  Determining projects to restore...
  All projects are up-to-date for restore.
/home/lev/CodeRepository/MyProjectSrvApi/MyProjectApi/Controllers/Pcaps.cs(12,29): error CS1061: 'PacketCapture' does not contain a definition for 'Packet' and no accessible extension method 'Packet' accepting a first argument of type 'PacketCapture' could be found (are you missing a using directive or an assembly reference?) [/home/lev/CodeRepository/MyProjectSrvApi/MyProjectApi/MyProjectApi.csproj]
/home/lev/CodeRepository/MyProjectSrvApi/MyProjectApi/Controllers/Pcaps.cs(18,32): error CS0246: The type or namespace name 'CaptureFileReaderDevice' could not be found (are you missing a using directive or an assembly reference?) [/home/lev/CodeRepository/MyProjectSrvApi/MyProjectApi/MyProjectApi.csproj]

**Comment by kayoub5** [comment]
Change `e.Packet` to `e.GetPacket().GetPacket()`

**Comment by lev99999** [comment]
Thank You!!
now it's work.
Can you explain me, please, why it's different from the tutorial?
Thanks

**Reply by kayoub5** [reply]
Tutorial is for version 5, for version 6 check the example projects
